src/core/thread/neighbor_table.hpp

```diff
@@ -85,75 +85,89 @@ class NeighborTable : public InstanceLocator
      * address.
      *
      * @param[in]  aShortAddress  A short address.
+     * @param[in]  aFilter        A neighbor state filter
```

Comment:
```suggestion
     * @param[in]  aFilter        A Neighbor state filter
```
I think `Neighbor` is a Thread network concept. 

Same issue for many other places.